File: Java_Backend/src/main/java/com/netlink/rsk/model/MonitorQuestionsVerification.java
package com.netlink.rsk.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Data
@Entity
@Table(name = "Monitorquestionsverification", catalog = "Training")
public class MonitorQuestionsVerification implements Serializable {

    @Id
    @Column(name = "Questionverificationid", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long questionVerificationId;

    @OneToOne
    @JoinColumn(name="Monitorquestionid", referencedColumnName = "Monitorquestionid")
    private MonitorQuestions monitorQuestions;

    @OneToOne
    @JoinColumn(name="Trainingid",referencedColumnName = "Trainingid")
    private Training training;

    @OneToOne
    @JoinColumn(name="Trainerid",referencedColumnName = "Trainerid")
    private TrainerMaster trainerMaster;

    @OneToOne
    @JoinColumn(name="Batchid",referencedColumnName = "Batchid")
    private BatchMaster batchMaster;

    @Column(name="Answer")
    private Boolean answer;

    @Column(name = "Createdby")
    private Integer createdBy;

    @Column(name = "Createdon")
    private Date createdOn;

    @OneToOne
    @JoinColumn(name = "Updatedby", referencedColumnName = "Userid")
    private UserMaster updatedBy;

    @Column(name = "Updatedon")
    private Date updatedOn;
}

File: client/src/app/layouts/training/model/monitor-questions-verification.ts
import { User } from "src/app/views/pages/login/model/user.model";
import { BatchMaster } from "./batch-master";
import { MonitorQuestions } from "./monitor-questions";
import { TrainerMaster } from "./trainer-master";
import { Training } from "./training";

export class MonitorQuestionsVerification {
    questionVerificationId: number;
    monitorQuestions: MonitorQuestions;
    training: Training;
    trainerMaster: TrainerMaster;
    batchMaster: BatchMaster;
    answer: boolean;
    createdBy: number;
    createdOn: Date;
    updatedBy: User;
    updatedOn: Date;
}
